kernel-open/nvidia/nv-pci.c

```diff
@@ -223,7 +223,79 @@ static void nv_init_dynamic_power_management
 
 
 
+static int nv_resize_pcie_bars(struct pci_dev *pci_dev) {
+    struct pci_host_bridge *host;
+    u16 cmd;
+    int r, old_size, requested_size;
+    int ret = 0;
+
+    // Check if BAR1 has PCIe rebar capabilities
+    u32 sizes = pci_rebar_get_possible_sizes(pci_dev, NV_GPU_BAR1);
+    if (sizes == 0) {
+        /* ReBAR not available. Nothing to do. */
+        return 0;
+    }
+
+    /* Try to resize the BAR to the largest supported size */
+    requested_size = fls(sizes) - 1;
+
+    /* If the kernel will refuse us, don't even try to resize,
+       but give an informative error */
+    host = pci_find_host_bridge(pci_dev->bus);
```

Comment:
pci_find_host_bridge() was merged in v5.15 by commit 59dc33252ee77, so this needs another conftest.sh entry. In this case, the implementation of pci_resize_resource() already checks host->preserve_config, so on kernels where this function is not exported I think we can safely skip this check here by surrounding it with conftest ifdefs.